lib/ansible/plugins/strategy/__init__.py

```diff
@@ -469,10 +469,12 @@ def search_handler_blocks_by_name(handler_name, handler_blocks):
                         self._tqm._failed_hosts[original_host.name] = True
 
                     if state and iterator.get_active_state(state).run_state == iterator.ITERATING_RESCUE:
+                        original_task_vars = self._variable_manager.get_vars(play=iterator._play, task=original_task)
+                        templar = Templar(loader=self._loader, variables=original_task_vars)
                         self._variable_manager.set_nonpersistent_facts(
                             original_host,
                             dict(
-                                ansible_failed_task=original_task.serialize(),
+                                ansible_failed_task=templar.template(original_task.serialize()),
```

Comment:
why not just use the actual task? since it is already a templated version of the original task

Reply:
this can also create divergences since the task itself might have changed variables it originally used:

``` set_fact: myvar={{ myvar +1 }}```

Reply:
> why not just use the actual task? since it is already a templated version of the original task

Do you mean like the following?
```
ansible_failed_task=original_task,
```

That would result in being different data structure than currently: e.g. `ansible_failed_task.role` vs. `ansible_failed_task._role` though since `serialize()` renames some of the attributes and "printing" `ansible_failed_task` would result in `task.name()` instead of the whole ds. Not sure if that's a problem though.

Also, in my testing, `var2` is not templated that way:
```
TASK [debug] ********************************************************************************************************************************
ok: [localhost] => {
    "ansible_failed_task._role._default_vars": {
        "var1": 1, 
        "var2": "{{ var1 }}"
    }
}
```

Reply:
no,  iirc at that point you have `task_result._task` which is 'already templated and same structure' as you need , no need to retemplate, specially since vars CAN change and you might get diff results

Reply:
> no, iirc at that point you have task_result._task which is 'already templated and same structure' as you need

But my previous comment shows otherwise, `var2` is not templated, the structure is different? I have used `original_task` but that is the same object as `task_result._task`.

> no need to retemplate, specially since vars CAN change and you might get diff results

I understand why we can't re-template.

Reply:
ah, nvmd, we overwrite the _task with original_task anyways, need to capture it before that happens